### ISSUE TITLE
如何设置一些special token，在计算loss的时候这些token的权重大点

### ISSUE THREAD

**Issue by wade0604** [modelscope/ms-swift#2605]
我想在qwen2中设置一些special token，然后在计算loss的时候设置这些token的loss scale为2，请问应该怎么操作呢

**Comment by Jintao-Huang** [comment]
可以设置plugin/loss_scale.py

**Comment by wade0604** [comment]
> 可以设置plugin/loss_scale.py
有几个问题想请假下：
1. 请问是只有agent的数据集可以使用loss scale参数吗？
2.假设我的special token =["<s1>","<s2>"]，loss_scale_map 这个json文件应该怎么配置，有demo可以参考吗？

**Comment by wade0604** [comment]
@Jintao-Huang 您好我看到ms-swift3.0[更新说明](https://github.com/modelscope/ms-swift/blob/main/docs/source/Instruction/ReleaseNote3.0.md)中提到支持定制每个token的loss权重，想问下这一块有具体的案例实现吗

**Comment by RomanticGodVAN** [comment]
@wade0604 请问设置好了吗，我最近也在看如何修改 loss_scale

**Comment by Gaotianhong** [comment]
我也有这个问题，按照 agent 的方式设置，会出现 shift_scale 和 masks 维度不匹配的现象

@register_loss_func(LossType.loss_scale)
def loss_scale_func(outputs, labels, loss_scale=None, num_items_in_batch=None) -> torch.Tensor:
    """Loss func

    Args:
        outputs: The model outputs
        labels: The labels
        loss_scale: The loss scale
        num_items_in_batch: Number of tokens in the labels of gradient accumulation round that are not -100.

    Returns:

    """
    loss, masks = ce_loss_func(outputs, labels)
    if loss_scale is not None:
        shift_scale = loss_scale[..., 1:].to(masks.device)
        shift_scale = shift_scale[masks]
        loss = (shift_scale * loss)
    if num_items_in_batch is None:
        loss = loss.mean()
    else:
        # compat transformers>=4.46
        loss = loss.sum() / num_items_in_batch
    return loss

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.

**Comment by wade0604** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
